File: msglib-csharp/MsgLib/Proto/MemoryDataBuffer.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace MsgLib
{
    internal class MemoryDataBuffer : IDataBuffer
    {
        internal MemoryStream Data { get; set; }

        internal MemoryDataBuffer()
        {
            this.Data = new MemoryStream();
        }
		internal MemoryDataBuffer(byte[] data)
			: this(data, false)
		{
		}
        internal MemoryDataBuffer(byte[] data, bool writable )
        {
            this.Data = new MemoryStream(data, writable);
        }

        internal MemoryDataBuffer(MemoryStream data)
        {
            this.Data = data;
        }

        int IDataBuffer.Read(byte[] buffer, int offset, int length)
        {
            return Data.Read(buffer, offset, length);
        }

        void IDataBuffer.Write(byte[] buffer, int offset, int length)
        {
            Data.Write(buffer, offset, length);
        }
    }
}

File: msglib-java/src/msglib/proto/io/MemoryDataBuffer.java
package msglib.proto.io;

import java.io.IOException;

import msglib.stream.MemoryStream;

public class MemoryDataBuffer extends MemoryStream implements IDataBuffer {

    public MemoryDataBuffer() {
    }

    public MemoryDataBuffer(byte[] data, boolean writable) {
        super(data, writable);
    }

    public MemoryDataBuffer(byte[] data, int offset, int length, boolean writable) {
        super(data, offset, length, writable, false);
    }

    @Override
    public int Read(byte[] buffer, int offset, int length) throws IOException {
        return super.read(buffer, offset, length);
    }

    @Override
    public void Write(byte[] buffer, int offset, int length) throws IOException {
        super.write(buffer, offset, length);
    }

}
